applicationset/generators/generator_spec_processor.go

```diff
@@ -81,3 +94,31 @@ func mergeGeneratorTemplate(g Generator, requestedGenerator *argoprojiov1alpha1.
 
 	return *dest, err
 }
+
+// Currently for Matrix Generator. Allows interpolating the matrix's 2nd child generator with values from the 1st child generator
+// "params" parameter is an array, where each index corresponds to a generator. Each index contains a map w/ that generator's parameters.
+// Since the Matrix generator currently only allows 2 child generators, effectively this params array will always be size 1 (containing the 1st child generator's params)
+// Uses similar process for interpreting values as the actual Application Template
+func interpolateGenerator(requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, params map[string]string) (argoprojiov1alpha1.ApplicationSetGenerator, error) {
+	interpolatedGenerator := requestedGenerator.DeepCopy()
+	tmplBytes, err := json.Marshal(interpolatedGenerator)
+	if err != nil {
+		log.WithError(err).WithField("requestedGenerator", interpolatedGenerator).Error("error marshalling requested generator for interpolation")
+		return *interpolatedGenerator, err
```

Comment:
Can we just return `nil, err`? I don't think we end up using the interpolatedGenerator's value if there's an error (we hit the `continue` instead and skip interpolation).

Reply:
That was actually my original plan (to use pointers). What we noticed is that in a scenario where you have say, two git generator paths matrixed w/ a cluster generator w/ the placeholder parameter, on the first pass when using a pointer it replaced the placeholder with the actual value altogether. So a label like this ` kubernetes.io/environment: '{{path.basename}}' ` was replaced by say `dev` on the 1st pass. So on the second pass, it would try to interpolate ` kubernetes.io/environment: dev `, but since the placeholder was already replaced, it would skip it altogether.